lib_eio_linux/eio_stubs.c

```diff
@@ -33,10 +34,16 @@ CAMLprim value caml_eio_mkdirat(value v_fd, value v_path, value v_perm) {
 CAMLprim value caml_eio_getrandom(value v_ba, value v_off, value v_len) {
   CAMLparam1(v_ba);
   ssize_t ret;
-  void *buf = Caml_ba_data_val(v_ba) + Long_val(v_off);
-  caml_enter_blocking_section();
-  ret = getrandom(buf, Long_val(v_len), 0);
-  caml_leave_blocking_section();
-  if (ret == -1) uerror("getrandom", Nothing);
+  size_t off = (size_t)Long_val(v_off);
+  size_t len = (size_t)Long_val(v_len);
+  while (off < len) {
```

Comment:
Why if `off < len`? It's perfectly reasonable to ask for e.g. len=2 bytes at offset=10.

I suggest writing the loop as `do { ... } while (ret == -1 && errno == EINTR)`.

(and `Flow.read_exactly` will handle repeating the read until the buffer is full, so no need to handle that here)

Reply:
Excellent suggestion! Done.